rsp/runtimes/bundles/org.jboss.tools.rsp.server.generic/src/main/java/org/jboss/tools/rsp/server/generic/servertype/actions/EditServerConfigurationActionHandler.java

```diff
@@ -0,0 +1,174 @@
+		WorkflowResponse workflow = new WorkflowResponse();
+		ServerActionWorkflow action = new ServerActionWorkflow(
+				ACTION_ID, ACTION_LABEL, workflow);
+		
+		List<WorkflowResponseItem> items = new ArrayList<>();
+		workflow.setItems(items);
+
+		WorkflowPromptDetails prompt = new WorkflowPromptDetails();
+		prompt.setResponseSecret(false);
+		prompt.setResponseType(ServerManagementAPIConstants.ATTR_TYPE_STRING);
+		prompt.setValidResponses(choices);
+		
+		// Simple action entirely on the UI side
+		WorkflowResponseItem item1 = new WorkflowResponseItem();
+		item1.setItemType(ServerManagementAPIConstants.WORKFLOW_TYPE_PROMPT_SMALL);
+		item1.setPrompt(prompt);
+		item1.setId(ACTION_EDIT_FILE_PROMPT_ID);
+		item1.setLabel(ACTION_EDIT_FILE_PROMPT_LABEL);
+		
+		items.add(item1);
+		workflow.setStatus(StatusConverter.convert(
+				new Status(IStatus.INFO, Activator.BUNDLE_ID, ACTION_LABEL)));
+		return action;
+	}
+
+	private ServerActionWorkflow cancelWorkflow() {
+		WorkflowResponse workflow = new WorkflowResponse();
+		ServerActionWorkflow action = new ServerActionWorkflow(
+				ACTION_ID, ACTION_LABEL, workflow);
+		
+		List<WorkflowResponseItem> items = new ArrayList<>();
+		workflow.setItems(items);
+
+
+		workflow.setStatus(StatusConverter.convert(
+				new Status(IStatus.CANCEL, Activator.BUNDLE_ID, ACTION_LABEL)));
+		return action;
+	}
+
+	protected ServerActionWorkflow executePath(String relative) {
+		String home = getDefaultWorkingDirectory();
+		IPath tmpPath = new Path(home).append(relative);
+		if(!tmpPath.toFile().isFile()) {
+			return cancelWorkflow();
+		}
+		
+		// Simple action entirely on the UI side
+		WorkflowResponseItem item1 = new WorkflowResponseItem();
+		item1.setItemType(ServerManagementAPIConstants.WORKFLOW_TYPE_OPEN_EDITOR);
+		Map<String,String> propMap = new HashMap<>();
+		propMap.put(ServerManagementAPIConstants.WORKFLOW_EDITOR_PROPERTY_PATH, tmpPath.toOSString());
+		item1.setProperties(propMap);
+		item1.setId(ACTION_ID);
+		item1.setLabel(ACTION_LABEL);
+
+		WorkflowResponse workflow = new WorkflowResponse();
+		ServerActionWorkflow action = new ServerActionWorkflow(
+				ACTION_ID, ACTION_LABEL, workflow);
+		
+		List<WorkflowResponseItem> items = new ArrayList<>();
+		workflow.setItems(items);
+
+
+		items.add(item1);
+		workflow.setStatus(StatusConverter.convert(
+				new Status(IStatus.OK, Activator.BUNDLE_ID, ACTION_LABEL)));
+		return action;
+	}
+	
+
+	protected String[] getConfigurationFilesRelative() {
+		String configFiles = "";
+		JSONMemento actions = this.genericServerDelegate.getActionsJSON();
+		if (actions != null) {
+			JSONMemento editConfigAction = actions.getChild("editServerConfiguration");
+			if (editConfigAction != null) {
+				configFiles = editConfigAction.getString("paths");
+			}			
+		}
+		if( configFiles == null || configFiles.length() == 0 )
+			return null;
+		String[] filesArr = configFiles.split(",");
```

Comment:
the same as the other split. isn't better split(",", 0)?

Reply:
split(",") is equivalent to split(",", 0)  so it is not necessary to change here.